### PR TITLE
jxl_threads: Hide all the std namespace symbols.

### PR DESCRIPTION
**Pull request by malaterre**
Closes issue (#1017)

### CODE REVIEW

File: lib/jxl_threads.cmake [hunk 1]
```diff
@@ -74,6 +74,16 @@ set_target_properties(jxl_threads PROPERTIES
   LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}"
   RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}")
 
+  set_target_properties(jxl_threads PROPERTIES
+      LINK_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/jxl/jxl_threads.version)
+  if(APPLE)
```
Comment (veluca93) [inline lib/jxl_threads.cmake:79]: I understand little about this, but the non-threads version has something for OSX here - why is an equivalent line not here?

**Review by veluca93 — DISMISSED** [review]

File: lib/jxl_threads.cmake (hunk 1 above)
Comment (malaterre) [reply lib/jxl_threads.cmake:79]: > I understand little about this, but the non-threads version has something for OSX here - why is an equivalent line not here?

Pay attention that I only tested the UNIX portion of the patch, no OSX system here.

Comment (malaterre) [reply lib/jxl_threads.cmake:79]: Also someone *must* review my patch. This is not entirely clear what the long term is with regards to `jxl_threads` vs simply `jxl`. Someone may suggest this is fine to use a single copy of this file (ie. re-use `jxl.version` against `jxl_threads`).

Comment (veluca93) [reply lib/jxl_threads.cmake:79]: Yup, I added the CI:full label which should trigger a run on OSX too.

As for reusing jxl.version - can you see any reason not to? In principle I'd be in favour of reusing it if possible.

Comment (malaterre) [reply lib/jxl_threads.cmake:79]: > As for reusing jxl.version - can you see any reason not to? In principle I'd be in favour of reusing it if possible.

I was influenced by these for the decision:

```
% find . | grep _export
./obj-x86_64-linux-gnu/lib/include/jxl/jxl_threads_export.h
./obj-x86_64-linux-gnu/lib/include/jxl/jxl_export.h
```

which explains why I picked the original solution...anyway I've updated my patch.

**Review by veluca93 — APPROVED** [review]

**Review by veluca93 — APPROVED** [review]

**Pull request merged**